packages/Search/src/DTK_DistributedSearchTree_decl.hpp

```diff
@@ -107,27 +103,25 @@ class DistributedSearchTree
 template <typename DeviceType>
 template <typename Primitives>
 DistributedSearchTree<DeviceType>::DistributedSearchTree(
-    Teuchos::RCP<Teuchos::Comm<int> const> comm, Primitives const &primitives )
+    MPI_Comm comm, Primitives const &primitives )
     : _comm( comm )
     , _bottom_tree( primitives )
 {
-    int const comm_rank = _comm->getRank();
-    int const comm_size = _comm->getSize();
+    int comm_rank;
+    MPI_Comm_rank( _comm, &comm_rank );
+    int comm_size;
+    MPI_Comm_size( _comm, &comm_size );
 
     Kokkos::View<Box *, DeviceType> boxes(
         Kokkos::ViewAllocateWithoutInitializing( "rank_bounding_boxes" ),
         comm_size );
-    // FIXME: I am not sure how to do the MPI allgather with Teuchos for data
-    // living on the device so I copied to the host.
+    // FIXME when we move to MPI with CUDA-aware support, we will not need to
```

Comment:
Is this just a matter of setting this up on the CI? Only open-mpi seems to support this right now.

Reply:
We would probably need to check at configure time that CUDA support is enabled.

Reply:
All the main mpi implementations support cuda.